tests/core/unittests/mongodb_test.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Collection of tests for :mod:`metaopt.io.database.mongodb`."""
+
+from copy import deepcopy
+
+from pymongo import MongoClient
+import pytest
+import yaml
+
+from metaopt.io.database.abstractdb import DatabaseError
+from metaopt.io.database.mongodb import MongoDB
+
+
+@pytest.fixture(scope='module')
+def exp_config():
+    """Load an example for an experiment entry."""
+    with open('./experiment.yaml') as f:
+        exp_config = yaml.safe_load(f)
+    return exp_config
+
+
+@pytest.fixture(scope='module')
+def database(exp_config):
+    """Return Mongo database object to test with example entries."""
+    conn = MongoClient(username='user', password='pass')
+    db = conn.metaopt_test
+    experiments = db.experiments
+    experiments.drop()
+    experiments.insert_one(exp_config)
+    yield db
+    db.close()
+
+
+@pytest.fixture(scope='function')
+def moptdb():
+    """Return MongoDB wrapper instance initiated with test opts."""
+    moptdb = MongoDB(dbname='metaopt-test', username='user', password='pass')
+    return moptdb
+
+
+def test_bad_connection():
+    """Raise when connection cannot be achieved."""
+    with pytest.raises(DatabaseError) as exc_info:
+        MongoDB(dbhost='asdfada', username='uasdfaf', password='paasdfss')
```

Comment:
How do you test this? You don't have a database right?

Reply:
So for the wrong host we would assert that the connection will definitely fail.

Reply:
Yes I was confused because I didn't know we had a mongo database setup for the tests. I realized when looking at the CI setup files.